Implement new “site cannot be accessed” flow

delan opened the pull request:
#### Changes proposed

This pull request implements the new call-to-action and flow for the “site cannot be accessed” indicator on non-Atomic sites (as described in [this](https://wp.me/p6TEKc-3DS-p2) [internal](https://wp.me/p6TEKc-3DS-p2#comment-7644) [post](https://wp.me/p6TEKc-3DS-p2#comment-7588)).

On such sites, the CTA changes to “I’d like to fix this now”, and it points to a new page that replaces the disconnect survey with some options that are more helpful for the situation. The indicator on Atomic sites and the disconnect button on the Manage Connection page remain unchanged.

There are some differences between these changes and [the design](https://wp.me/p6TEKc-3DS-p2#comment-7644), as well as some known problems, so I’ve added comments for them below. Let me know if these changes need any automated tests.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

##### 1. Site indicator

1. Open [this branch on calypso.live](https://calypso.live/?branch=add/site-cannot-be-accessed-flow), click **My Sites**, then click **Switch Site** if necessary
1. If you have any Atomic Jetpack sites with the red exclamation mark, click that icon, verify that the call-to-action reads **Remove Site**, click that link, then go to step 2.3
1. If you have any non-Atomic Jetpack sites with the red exclamation mark, click that icon, verify that the call-to-action reads **I’d like to fix this now**, click that link, then go to step 3.3

##### 2. Old page

1. Open [this branch on calypso.live](https://calypso.live/?branch=add/site-cannot-be-accessed-flow), click **My Sites**, then click **Switch Site** if necessary
1. Click any Jetpack site, go to **Manage** > **Settings** > **Manage your connection**, then click **Disconnect from Wordpress.com**
1. Verify that you see the old disconnect survey, then click **Skip for now**
1. Verify that you see the **Confirm Disconnection** page, then click **Back**
1. Verify that you see the old disconnect survey

##### 3. New page

1. Open [this branch on calypso.live](https://calypso.live/?branch=add/site-cannot-be-accessed-flow), click **My Sites**, then click **Switch Site** if necessary
1. Click any Jetpack site, go to **Manage** > **Settings** > **Manage your connection** > **Disconnect from Wordpress.com**, then append `?type=down` to the end of your URL
1. Verify that you see the new page, then click **Confirm your site is loading**
1. Verify that the site opens in a new tab or window, close it, then click **Troubleshoot Jetpack**
1. Verify that the **Fixing Jetpack Connection Issues** page opens in the same tab or window, go back, then click **Disconnect Jetpack**
1. Verify that you see the **Confirm Disconnection** page, then click **Back**
1. Verify that you see the new page

#### Screenshots

* [new call-to-action under All My Sites](https://user-images.githubusercontent.com/465303/83127615-b9b51f80-a11d-11ea-8150-a782b9d71fb6.png)
* [new call-to-action under a single site](https://user-images.githubusercontent.com/465303/83127752-e23d1980-a11d-11ea-8957-befc7257ed45.png)
* [new page](https://user-images.githubusercontent.com/465303/83019782-dcd2c700-a06a-11ea-8ecb-c66413f8c776.png)

delan commented on client/my-sites/site-settings/disconnect-site/index.jsx:
```diff
@@ -9,28 +9,123 @@ import { localize } from 'i18n-calypso';
+						) }
+					/>
+					<div className="disconnect-site__actions">
+						<CompactCard href={ siteUrl } target="_blank" rel="noopener noreferrer">
```
design: unlike the figma prototype, these are link-coloured and normal font-weight, is that ok?

delan replied:
(i’ve used href cards here because all of these options have a URL, so i think we should let users do anything they can do with other links (middle-click new tab, right-click bookmark, …), but we can definitely override the appearance or do this another way)

keoshi replied:
Thanks for this change! Having the link color there actually gives this more separation between the options and make them stand out a bit more. Let's go with this.

delan commented on client/my-sites/site-settings/disconnect-site/index.jsx:
```diff
@@ -9,28 +9,123 @@ import { localize } from 'i18n-calypso';
+} ) => {
+	const confirmHref = `/settings/disconnect-site/confirm/${ siteSlug }?type=down`;
 
 	let backHref = '/settings/manage-connection/' + siteSlug;
```
bug: the Back buttons on both old and new pages always take you back to Manage Connection (don’t worry about the reason branch below, it looks like a red herring), but if you used the CTA, that’s not necessarily where you came from

delan replied:
should i make this onClick history.back + preventDefault and keep the href as a fallback, or is there a way to get the previous URL out of the router and set href to that?

keoshi replied:
I would be pretty cool if we can actually make it go back in history rather than dropping them in the Manage Connection page. Even though they won't be able to do much since the connection is broken, it would be nice that all the `<- Back` links would behave in an expectable way.

jsnajdr replied:
A simple
```js
import page from 'page';
page.back();
```
should be enough to go back.

Or `window.history.back()`. The difference is that `page.back()` will never navigate out of Calypso (i.e., to another page that was previously opened in the same tab.

If you want to construct a full `<a href="..."/>` with a valid link, then the `getPreviousRoute( state )` Redux selector will return the previous URL.

All of the above will work only if there is some navigation history to work with. If you got to the page by opening a direct link (from email, from chat, ...), clicking on the "back" arrow will do nothing.

delan replied:
wow there are more options than i realised, and we could combine them in several ways, plus there’s also the question of whether this NavigationLink should go back through history (enabling the browser forward button) or push forward to a new history entry… does calypso have a convention i can defer to here?

delan replied:
![child laughing then crying](https://user-images.githubusercontent.com/465303/84477282-b94a8600-acd2-11ea-88b3-7605a08fe753.png)

keoshi commented on client/my-sites/site-settings/disconnect-site/style.scss:
```diff
@@ -17,6 +17,60 @@
+	}
+
+	.disconnect-site__confirm-icon {
+		color: var( --color-primary );
```
Thanks for bringing it up. Testing it just now I feel `--studio-blue-40` is the best one, since it brings all the icons to the same level of brightness.

delan replied:
done in dc3aaefb6e062f94e37bfbfe8668dfac049eecff (i’ve used `--color-primary-40` based on the [colour docs](https://wpcalypso.wordpress.com/devdocs/docs/color.md#semantic-colors) guidance, but let me know if that’s no good)

keoshi commented on client/my-sites/site-settings/disconnect-site/style.scss:
```diff
@@ -17,6 +17,60 @@
+	}
+
+	.disconnect-site__troubleshoot-icon {
+		color: var( --color-jetpack );
```
The logo color (`#00be28`) is slowly being replaced by the palette that the Color Studio gives us, so let's go with JP Green 30 here.

keoshi commented on client/my-sites/site-settings/disconnect-site/style.scss:
```diff
@@ -17,6 +17,60 @@
+	}
+
+	.disconnect-site__disconnect-icon {
+		color: var( --color-error-40 );
```
Sorry about all the color discrepancies here. You made all the right choices though: Error 40 looks great!

delan replied:
no worries!

keoshi commented on client/my-sites/site-settings/disconnect-site/index.jsx:
```diff
@@ -9,28 +9,123 @@ import { localize } from 'i18n-calypso';
+					<DocumentHead title={ translate( 'Site Settings' ) } />
+					<FormattedHeader
+						headerText={ translate(
+							'Your site {{strong}}%(siteName)s{{/strong}} cannot be accessed',
```
The difference in weight in this typeface is more subtle than desirable, but yes — let's keep it! Nice tweak.

sirreal commented on client/my-sites/site-settings/disconnect-site/index.jsx:
```diff
@@ -9,28 +9,125 @@ import { localize } from 'i18n-calypso';
 	}
 
+	if ( type === 'down' ) {
+		confirmHref += '?type=down';
```
Do we have any guarantees about exactly what `confirmHref` is? It may be better to use `addQueryArgs` for this one.

delan replied:
good point! only in the current state of the component, but not if new code changes confirmHref, so i’ve fixed that in af627f3b90b028c6ff62b78adb4b33696133e273

jsnajdr commented on client/my-sites/site-indicator/index.jsx:
```diff
@@ -205,17 +205,22 @@ class SiteIndicator extends Component {
-						href={ `/settings/disconnect-site/${ site.slug }` }
+						href={ `/settings/disconnect-site/${ site.slug }${
+							siteIsAutomatedTransfer ? '' : '?type=down'
+						}` }
```
The `/settings/disconnect-site/:site` route won't do anything useful for an Atomic site. It will [redirect to `/settings/general`](https://github.com/Automattic/wp-calypso/blob/909abf45b46a7931d4b54826de11b0eae37a93bd/client/my-sites/site-settings/redirect-non-jetpack/index.jsx#L35-L37).

delan replied:
fixed in bececfbe0b8812be9a79c3d8897afe02e783c6c2

jsnajdr commented on client/my-sites/site-settings/disconnect-site/index.jsx:
```diff
@@ -9,28 +9,125 @@ import { localize } from 'i18n-calypso';
+		confirmHref += '?type=down';
+
+		return (
+			<div>
```
This extra `div` around the `Main` component seems to be redundant. That's true also for the existing markup in this component.

delan replied:
i thought that might be the case! done in ae5d99c9a6cb2abeeb59f8daae2b175f203bcdae

jsnajdr commented on client/my-sites/site-settings/disconnect-site/index.jsx:
```diff
@@ -9,28 +9,125 @@ import { localize } from 'i18n-calypso';
 		backHref = '/settings/disconnect-site/' + siteSlug;
 	}
 
+	if ( type === 'down' ) {
```
This big `if () { ... } else { ... }` block could be split into two React components.

delan replied:
done in ae5d99c9a6cb2abeeb59f8daae2b175f203bcdae

jsnajdr commented on client/my-sites/site-settings/disconnect-site/index.jsx:
```diff
@@ -52,6 +149,7 @@ const DisconnectSite = ( { reason, siteSlug, translate } ) => {
 
 const connectComponent = connect( ( state ) => ( {
 	siteSlug: getSelectedSiteSlug( state ),
+	site: getSelectedSite( state ),
```
Now when we need more info about the site than just the slug, we can call only `getSelectedSite` and access the slug as `site.slug`.

delan replied:
done in 2a7c839ae38cc7a197af66df24bc202f7e3e9a18

delan commented:
Thanks for the feedback @keoshi + @sirreal + @jsnajdr! I’ve got a jpop-issues rotation next week, but I’ll try to work my way through everything over the next few days.

p.s. I’m such a big fan of real quotation marks and other fancy Unicode punctuation that I use the Mac keyboard layout [on my Linux machines](https://github.com/delan/config/blob/67a167e97f81b34be60e56e35e6f690a01c1e655/nix/nixos/lib/interactive.nix#L31), and I’m hoping I can [do the same](https://www.microsoft.com/en-us/download/details.aspx?id=22339) on Windows too

delan commented on client/my-sites/site-indicator/index.jsx:
```diff
@@ -205,17 +205,22 @@ class SiteIndicator extends Component {
-						{ translate( 'Remove Site' ) }
+						{ siteIsAutomatedTransfer
+							? translate( 'Remove Site' )
+							: translate( 'I’d like to fix this now' ) }
```
interesting! i must admit i’m not that familiar with Atomic, so i’ve only been going off @keoshi’s note on the internal post (below) and [the field guide](https://wp.me/PCYsg-bTg-p2). should we double check with someone else?

> Note: Atomic sites should still see the existing messaging/link.

jsnajdr replied:
The situation with Atomic sites is very interesting and I'm also curious what "disconnecting from Jetpack" means for them. I posted a question in the #atomic Slack channel.

keoshi replied:
Link to that discussion in case you're interested in following along: p1591012708323500-slack-atomic

delan replied:
fixed in bececfbe0b8812be9a79c3d8897afe02e783c6c2

delan commented:
@keoshi 

I’ve fixed image 1 in ef6db93ab988412e5d89260a3b77938773c42ff9, and fixed image 2 in 1362f7f7033dba7a20915eec63726b341dc4e68d by setting the margin-left to 0 as suggested, though I’ve currently done that for all viewport sizes.

@sirreal

> If a Jetpack site is properly connected, there's no indication of that in the disconnect site view. This is an edge case and may not be worth addressing in this iteration, but an indication when a site appears to be properly connected would be nice.

For what it’s worth, the only way to reach the new disconnect flow (short of URL hacking) is via the call-to-action in a broken site indicator, but if that’s still a concern we could certainly put together a redirect or some alternate copy.

> The only support option seems to be a link to a documentation page. Even for sites with a page plan, an opportunity to contact Happiness seems to be missing.

I thought we could change this link to `https://jetpack.com/support/debug/?url=<site>` (same as **Manage Connection** > **Diagnose a connection problem**) at first, but neither page seems to provide a direct Happiness opportunity. Perhaps one for @keoshi?

sirreal commented:
> > If a Jetpack site is properly connected, there's no indication of that in the disconnect site view. This is an edge case and may not be worth addressing in this iteration, but an indication when a site appears to be properly connected would be nice.
> 
> For what it’s worth, the only way to reach the new disconnect flow (short of URL hacking) is via the call-to-action in a broken site indicator, but if that’s still a concern we could certainly put together a redirect or some alternate copy.

An observation I raised, I agree it's not a pressing issue and doesn't need to be addressed here.

> > The only support option seems to be a link to a documentation page. Even for sites with a page plan, an opportunity to contact Happiness seems to be missing.
> 
> I thought we could change this link to `https://jetpack.com/support/debug/?url=<site>` (same as **Manage Connection** > **Diagnose a connection problem**) at first, but neither page seems to provide a direct Happiness opportunity. Perhaps one for @keoshi?

Again, this was an observation and I didn't read up on all the context around the changes. I believe the experience here is as intended so you can disregard my comment.

jsnajdr commented on client/my-sites/site-settings/disconnect-site/index.jsx:
```diff
@@ -9,49 +9,40 @@ import { localize } from 'i18n-calypso';
-import FormattedHeader from 'components/formatted-header';
-import Main from 'components/main';
-import NavigationLink from 'components/wizard/navigation-link';
+import { DownFlow } from './down-flow';
```
Nit: it's more common to export React components as default exports, especially if there's only one in the file.

One practical benefit of that is that libraries that do lazy loading of React components usually assume a `default` export by default.

delan replied:
done in e91541c3e89649838c4a0bc766308f0a63ecec2f

jsnajdr commented on client/my-sites/site-settings/disconnect-site/index.jsx:
```diff
@@ -9,49 +9,40 @@ import { localize } from 'i18n-calypso';
+	if ( type === 'down' ) {
+		return (
+			<DownFlow
+				translate={ translate }
```
Instead of passing the `translate` prop down, it's more preferrable to use the `useTranslate` hook there:
```jsx
export default function DownFlow( props ) {
  const translate = useTranslate();
  ...
}
```

This is also a great opportunity to migrate the `<DisconnectSite />` component away from the `localize` HOC, to use the `useTranslate` hook.

delan replied:
done in 080554558bf4cb68c0109dcf844c8a8f555ce89f

michelleweber commented:
Some copy tweaks!
**Your site, NAME, can't be accessed.**
Jetpack wasn't able to connect to URL. Let's figure out why -- there are a few things to try.

**Confirm that your site loads.**
First, open your site to see if it loads properly.

**Troubleshoot Jetpack.**
If your site is working but you're still seeing this error, let's troubleshoot your Jetpack connection.

**Disconnect Jetpack.**
If you're no longer using Jetpack and/or WordPress for your site, or you've taken your site down, it's time to disconnect Jetpack.

Note: You may be seeing this error because you don't have admin permissions for this site. If you think you should, contact your site administrator.

------

One question: if I AM using JP and my site doesn't load, and I disconnect JP, how do I know what to do next?

keoshi commented on client/my-sites/site-settings/disconnect-site/down-flow.jsx:
```diff
@@ -0,0 +1,97 @@
+		<Main className="disconnect-site__down-flow">
+			<DocumentHead title={ translate( 'Site Settings' ) } />
+			<FormattedHeader
+				headerText={ translate( 'Your site {{strong}}%(siteName)s{{/strong}} cannot be accessed', {
```
```suggestion
				headerText={ translate( 'Your site ,{{strong}}%(siteName)s{{/strong}}, cannot be accessed', {
```

keoshi commented on client/my-sites/site-settings/disconnect-site/down-flow.jsx:
```diff
@@ -0,0 +1,97 @@
+					components: { strong: <strong /> },
+				} ) }
+				subHeaderText={ translate(
+					'Jetpack is unable to connect to your site at {{strong}}%(siteSlug)s{{/strong}}.{{br/}}There might be a few reasons for that — how do you want to proceed?',
```
```suggestion
					'Jetpack wasn’t able to connect to {{strong}}%(siteSlug)s{{/strong}}.{{br/}}Let’s figure out why — there are a few things to try.',
```

keoshi commented on client/my-sites/site-settings/disconnect-site/down-flow.jsx:
```diff
@@ -0,0 +1,97 @@
+					<FormattedHeader
+						isSecondary
+						align="left"
+						headerText={ translate( 'Confirm your site is loading' ) }
```
```suggestion
						headerText={ translate( 'Confirm that your site loads' ) }
```

keoshi commented on client/my-sites/site-settings/disconnect-site/down-flow.jsx:
```diff
@@ -0,0 +1,97 @@
+						align="left"
+						headerText={ translate( 'Confirm your site is loading' ) }
+						subHeaderText={ translate(
+							'As a first step, we suggest trying to open your site to ensure it’s loading correctly.'
```
```suggestion
							'First, open your site to see if it loads properly.'
```

keoshi commented on client/my-sites/site-settings/disconnect-site/down-flow.jsx:
```diff
@@ -0,0 +1,97 @@
+						align="left"
+						headerText={ translate( 'Troubleshoot Jetpack' ) }
+						subHeaderText={ translate(
+							'Select this option if you’re still using Jetpack but seeing this error.'
```
```suggestion
							'If your site is working but you’re still seeing this error, let’s troubleshoot your Jetpack connection.'
```

keoshi commented on client/my-sites/site-settings/disconnect-site/down-flow.jsx:
```diff
@@ -0,0 +1,97 @@
+						align="left"
+						headerText={ translate( 'Disconnect Jetpack' ) }
+						subHeaderText={ translate(
+							'Select this option if your site isn’t available anymore or if you’ve stopped using Jetpack and/or WordPress.'
```
```suggestion
							'If you’re no longer using Jetpack and/or WordPress for your site, or you’ve taken your site down, it’s time to disconnect Jetpack.'
```

keoshi commented on client/my-sites/site-settings/disconnect-site/down-flow.jsx:
```diff
@@ -0,0 +1,97 @@
+			</div>
+			<p>
+				{ translate(
+					'You might also be seeing this error because you don’t have enough permissions on this site.{{br/}}Please contact your site administrator for more details.',
```
```suggestion
					'Note: You may be seeing this error because you don’t have admin permissions for this site.{{br/}}If you think you should, contact your site administrator.',
```

michelleweber commented:
Got it. That makes sense, and it also makes me want to change the copy for the first point just a little bit to something like:

Confirm that your site loads.
Visit your site to make sure it loads properly. If there's an issue, fix your site before worrying about Jetpack! That may resolve this error.

delan commented:
Thanks for the feedback everyone! Once the PR is good to go and the builds are green, could someone please help me with the deployment side of things (maybe @sirreal or @jsnajdr)?

a8ci18n commented on client/my-sites/site-settings/disconnect-site/down-flow.jsx:
```diff
@@ -0,0 +1,103 @@
+					<FormattedHeader
+						isSecondary
+						align="left"
+						headerText={ translate( 'Disconnect Jetpack.' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **43** times:
`translate( 'Disconnect Jetpack' )` *ES Score: 12*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-cannot-be-accessed-flow)

keoshi commented:
Thanks for all those changes, @delan !

One question that popped in my head when looking at this was if it would be possible to add tracking to the different options before merging this PR?

a8ci18n commented on client/my-sites/site-settings/disconnect-site/down-flow.jsx:
```diff
@@ -0,0 +1,118 @@
+					<FormattedHeader
+						isSecondary
+						align="left"
+						headerText={ translate( 'Disconnect Jetpack.' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **43** times:
`translate( 'Disconnect Jetpack' )` *ES Score: 12*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-cannot-be-accessed-flow)

delan commented:
@nsakaimbo Thanks for the heads up! ci/wp-desktop is now green on this branch.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3815855

Hi @delan, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `I’d like to fix this now`
- `Your site, {{strong}}%(siteName)s{{/strong}}, cannot be accessed`
- `Jetpack wasn’t able to connect to {{strong}}%(siteSlug)s{{/strong}}.{{br/}}Let’s figure out why — there are a few things to try.`
- `Confirm that your site loads.`
- `Visit your site to make sure it loads properly. If there’s an issue, fix your site before worrying about Jetpack! That may resolve this error.`
- `Troubleshoot Jetpack.`
- `If your site is working but you’re still seeing this error, let’s troubleshoot your Jetpack connection.`
- `If you’re no longer using Jetpack and/or WordPress for your site, or you’ve taken your site down, it’s time to disconnect Jetpack.`
- `Note: You may be seeing this error because you don’t have admin permissions for this site.{{br/}}If you think you should, contact your site administrator.`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3815855).
